Mathlib/NumberTheory/JacobiSum/Basic.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2024 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.NumberTheory.MulChar.Basic
+import Mathlib.Algebra.Module.BigOperators
+
+/-!
+# Jacobi Sums
+
+This file defines the *Jacobi sum* of two multiplicative characters `χ` and `ψ` on a finite
+commutative ring `R` with values in another commutative ring `R'`:
+
+`jacobiSum χ ψ = ∑ x : R, χ x * ψ (1 - x)`
+
+(see `jacobiSum`) and provides some basic results and API lemmas on Jacobi sums.
+
+## References
+
+We essentially follow
+* [K. Ireland, M. Rosen, *A classical introduction to modern number theory*
+   (Section 8.3)][IrelandRosen1990]
+
+but generalize where appropriate.
+
+This is based on Lean code written as part of the bachelor's thesis of Alexander Spahl.
+-/
+
+open BigOperators Finset
+
+/-!
+### Jacobi sums: definition and first properties
+-/
+
+section Def
+
+-- Is this useful enough to be non-private?
```

Comment:
I think so. And I guess you can use `#find_home` to find a better place for it. And maybe generalize `1` to an arbitrary element `r` of `R`?
It could also be a `simp`-lemma, I think.

And if you like it, the version with `x + r` should be just as short, and can maybe be added for completeness?

Reply:
Further inspection shows that this is (up to DefEq / `Equiv.subLeft_apply`) just `(Equiv.subLeft 1).sum_comp f`, so I'm now opting for inlining it.